src/migrations/20221017000000-missing-objectives.js

```diff
@@ -0,0 +1,790 @@
+                  mg."firstInProgressAt",
+                  mg."lastInProgressAt",
+                  mg."firstCeasedSuspendedAt",
+                  mg."lastCeasedSuspendedAt",
+                  mg."firstClosedAt",
+                  mg."lastClosedAt",
+                  mg."firstCompletedAt",
+                  mg."lastCompletedAt"
+                FROM "temp_incomplete_reports" ir
+                JOIN "temp_missing_goals" mg
+                ON ir."activityReportId" = mg."activityReportId"
+                AND mg."goalHash" = ANY(ir."distinctGoals")
+                LEFT JOIN "Goals" g
+                ON g."grantId" = ANY(ir."onlyRecipientGrantId")
+                AND md5(g.name) = mg."goalHash"
+                WHERE g.id IS NULL
+                RETURNING
+                  id "goalId"
+              )
+              SELECT "goalId"
+              FROM "inserted";
+
+          -----------------------------------------------------------------------------------------------------
+          -- 4. Create new ActivityReportGoals for added goals
+          -----------------------------------------------------------------------------------------------------
+            DROP TABLE IF EXISTS "temp_missing_args_added";
+              CREATE TEMP TABLE "temp_missing_args_added" AS
+          WITH "inserted" AS (
+              INSERT INTO "ActivityReportGoals"
+              (
+                "activityReportId",
+                "goalId",
+                "createdAt",
+                "updatedAt"
+              )
+              SELECT
+                ir."activityReportId",
+                g.id "goalId",
+                MIN(arg2."createdAt") "createdAt",
+                MAX(arg2."updatedAt") "updatedAt"
+              FROM "temp_missing_goals_added" mga
+              JOIN "Goals" g
+              ON mga."goalId" = g.id
+              JOIN "temp_incomplete_reports" ir
+              ON g."grantId" = ANY(ir."onlyRecipientGrantId")
+              AND md5(g.name) = ANY(ir."distinctGoals")
+              LEFT JOIN "ActivityReportGoals" arg
+              ON ir."activityReportId" = arg."activityReportId"
+              AND g.id = arg."goalId"
+              LEFT JOIN "ActivityReportGoals" arg2
+              ON ir."activityReportId" = arg2."activityReportId"
+              LEFT JOIN "Goals" g2
+              ON arg2."goalId" = g2.id
+              AND g."grantId" = ANY(ir."goalGrantIds")
+              AND md5(g.name) = md5(g2.name)
+              WHERE arg.id IS NULL
+              GROUP BY
+                ir."activityReportId",
+                g.id
+          RETURNING
+            id "activityReportGoalId"
+          )
+          SELECT "activityReportGoalId"
+          FROM "inserted";
+              -----------------------------------------------------------------------------------------------------
+            DROP TABLE IF EXISTS "temp_goals_now_on_approved_ar";
+              CREATE TEMP TABLE "temp_goals_now_on_approved_ar" AS
```

Comment:
This doesn't appear to select anything.

Reply:
That is correct as the correct value was already aggerated while building the data, but it is still included to verbosely illustrate that this step is being checked and verified.